pandas/tests/io/excel/test_readers.py

```diff
@@ -1499,3 +1499,14 @@ def test_engine_invalid_option(self, read_ext):
         with pytest.raises(ValueError, match="Value must be one of *"):
             with pd.option_context(f"io.excel{read_ext}.reader", "abc"):
                 pass
+
+    def test_corrupt_files_closed(self, request, engine, read_ext):
+        # GH41778
+        with tm.ensure_clean(f"corrupt{read_ext}") as file:
+            with pytest.warns(None) as record:
+                try:
+                    with pd.ExcelFile(file, engine=engine) as _:
```

Comment:
Thanks @twoertwein for the update. can we test without the context manager too, to replicate the usage reported in the issue.

Reply:
I removed the context manager